hazelcast-integration/kubernetes/samples/external-client/README.md

```diff
@@ -0,0 +1,201 @@
+    service-per-pod-ports: "5701:5701"
+``` 
+
+### 3. Configure Service Account
+
+Hazelcast uses Kubernetes API for the member discovery and therefore it requires granting permission to certain resources. To create ServiceAccount with minimal permissions, run the following command.
+
+```
+kubectl apply -f rbac.yaml
+```
+
+The Service Account 'hazelcast-service-account' was created and you can use it in all further steps.
+
+### 4. Install Hazelcast cluster
+
+To install Hazelcast cluster, you need to include the Service-Per-Pod annotations into your StatefulSet (or Deployment) Hazelcast configuration. Then, deploy Hazelcast cluster into your Kubernetes environment.
+
+```
+kubectl apply -f hazelcast-cluster.yaml
+``` 
+
+You can check that for each Hazelcast Member POD there was a service created and that Hazelcast members formed a cluster.
+
+```
+$ kubectl get all
+NAME              READY     STATUS    RESTARTS   AGE
+pod/hazelcast-0   1/1       Running   0          2m
+pod/hazelcast-1   1/1       Running   0          1m
+pod/hazelcast-2   1/1       Running   0          1m
+
+NAME                  TYPE           CLUSTER-IP      EXTERNAL-IP      PORT(S)          AGE
+service/hazelcast-0   LoadBalancer   10.19.241.253   35.188.83.111    5701:30597/TCP   2m
+service/hazelcast-1   LoadBalancer   10.19.251.243   35.192.168.46    5701:32718/TCP   2m
+service/hazelcast-2   LoadBalancer   10.19.254.0     35.193.248.247   5701:30267/TCP   2m
+service/kubernetes    ClusterIP      10.19.240.1     <none>           443/TCP          1h
+
+$ kubectl logs pod/hazelcast-2
+...
+Members {size:3, ver:3} [
+        Member [10.16.1.10]:5701 - abab30fe-5a45-484d-bad5-e60c252572ca
+        Member [10.16.2.7]:5701 - 9b948e91-0115-470f-850e-d5cbf2e3b0e1
+        Member [10.16.0.8]:5701 - e68ce431-4000-467b-92c6-0072b2601d60 this
+]
+...
+```
+
+## Configure Hazelcast Client outside Kubernetes
+
+When we have a working Hazelcast cluster deployed on Kubernetes, we can connect to it with an external Hazelcast Smart Client. You need first to fetch the credentials of the created Service Account and then use them to configure the client.
+
+### 5. Check Kubernetes Master IP
+
+To check the IP address of the Kubernetes Master, use the following command.
+
+```
+$ kubectl cluster-info
+Kubernetes master is running at https://35.226.182.228
+```
+
+### 6. Check Access Token and CA Certificate
+
+First, you need to find the name of the secret for the created Service Account.
+
+```
+$ kubectl get secret
+NAME                                 TYPE                                  DATA      AGE
+default-token-q9sp8                  kubernetes.io/service-account-token   3         2h
+hazelcast-service-account-token-6s94h   kubernetes.io/service-account-token   3         9m
+```
+
+Then, to fetch Access Token, use the following command.
+
+```
+$ kubectl get secret hazelcast-service-account-token-6s94h -o jsonpath={.data.token} | base64 -d
```

Comment:
Interesting, i see Mac has a different base64 version than ubuntu ;) Changed to `--decode`, it should work for both.

Reply:
`--decode` option is not working for `alpine`:
```
bash-4.3# base64 --decode
base64: unrecognized option: decode
BusyBox v1.24.2 (2017-01-18 14:13:46 GMT) multi-call binary.

Usage: base64 [-d] [FILE]

Base64 encode or decode FILE to standard output
	-d	Decode data
```

Reply:
Yeah, we can conclude that there is no single option that would work for all: Mac, Ubuntu, and Alpine ;) 
Since the code sample is dedicated to humans and humans usually use Mac/Windows/Ubuntu, then I'd leave `--decode`.